### ISSUE TITLE
Submission wizard practice area

### ISSUE THREAD

**Issue by only1chunts** [gigascience/gigadb-website#72]
Aim: Submitters like to "have a play" before they start a real submission, we need a way to allow them to do that without clogging up the production database. This would also have the benefit that we could use it for teaching purposes.
Action: provide a "test"/mock submission wizard link within a users personal profile that they can use to findout what sort of information we ask for. Additionally, it would be nice to have the ability for users to "port" their practice submissions to the production version if they need to, could that be something the API can handle? For example, if a user gets to the point of uploading files, they should be asked if they intended this submission to be real rather than practice. No practice submissions should be allowed to make file uploads, and should not be allocated FTP disk space.

**Comment by pli888** [comment]
Hello @only1chunts and @jessesiu:

To implement the submission wizard practice area, @lump1k asks:

>do you agree that just add some field to dataset table like "is_test" and create cron task to clear all the test datasets will be the best decision?

**Comment by only1chunts** [comment]
Yes, when we discussed this on the call we agreed that's the easiest option. Some careful thought needs to put into how the delete cascade works, i.e just deleting 1 row from the dataset table is not sufficient if the test has included additional authors, samples, attributes, file, funding etc.

**Comment by pli888** [comment]
I agree with @only1chunts that implementing the submission wizard practice area needs careful thought. @lump1k it would be good if you can describe how you will implement the practice area functionality to us when you are ready to work on it.

**Comment by lump1k** [comment]
I added to dataset table 2 fileds: is_test and creation_date
Is test means is it test dataset or real, creation date i need command for clearing all test datasets which created 1 month ago and later (can be changed to 1 week or something like this if you want)

I added to attribute table 1  filed: is_test. Because in Sample table user have possibility to create attributes and if it a test dataset it will be bad idea to show these attributes for real datasets. So if user create some attribute for test dataset this attribute also will be test

For dataset model i created 2 methods: removeWithAllData (removes all authors, fundings, links, etc which related to this dataset, image if exist and this dataset) and toReal (makes this dataset real, is_test = 0 and also makes all test attributes as real)

And also i created command for cron to clear all test datasests TestDatasetsCommand.
You can run it like this
docker-compose run --rm  application ./protected/yiic testdatasets clear

**Comment by only1chunts** [comment]
Not sure why you think the user has the ability to add new attributes during the submission process? this is not a function that should be allowed, users must select from the set of attributes we already have in the ATTRIBUTES database table.
Do you mean the SAMPLE_ATTRIBUTE table? (which the users will be filling by adding attribute values during the submission process) I dont think there is any need to add column to that table because the test SAMPLE will be deleted with the test DATASET, so the test SAMPLE_ATTRIBUTES can also be deleted As well as the test DATASET_SAMPLE rows, and the test DATASET_AUTHORS, and any AUTHOR rows that are only used by test datasets, the level of complexity here is why I originally suggested a second test database.
Every database table that can be written to during the submission wizard process needs to be checked for removal of the test data, and I would rather not bloat the entire database with new is_test columns for every table.
Please explain more clearly how you intend to ensure all relevant objects are deleted when a test dataset is deleted?

**Comment by lump1k** [comment]
http://i.imgur.com/UcLbq87.png
as I remember I saw in powerpoint presentation that if user add some attribute which doesnt exist we should show him closest by name or create new. and i did it like this now. so user can create a lot of own attributes if he wants. But maybe I didnt get something. Should it be error message when he add some new attribute name?

**Comment by only1chunts** [comment]
I dont recall saying we should create a new attribute if the user wants
one, but...
it is on my wish list for a future task, but that also involves asking the
submitter for more details about the attribute so that its not just a
random meaningless name, has your implementation included gathering a
description of the new attribute and the units to be used and any
appropriate ontologies that should be used? If so, thats great and I look
forward to seeing it in action. If not, then we will probably need to
block/remove that option.

**Comment by lump1k** [comment]
no. it creates a new attribute only by name. so should I show error message (Attribute doesnt exist) in this case?
If in this page we dont create new atrributes we dont need is_test field in attribute table. I agree with you

**Comment by only1chunts** [comment]
yes, please create an error message that asks the user to either try an
alternative attribute name or use "miscellaneous parameter" and include
their own attribute name within the value, e.g. miscellaneous
parameter=users-own-attribute-name:value-of-attribute
this will insert the column header as "miscellaneous parameter", and the
values will be two parts separated by a colon "name:value".
